File: app/src/main/java/com/marknkamau/justjava/activities/profile/ProfileActivityView.java
package com.marknkamau.justjava.activities.profile;

import com.marknkamau.justjava.models.PreviousOrder;

import java.util.List;
import java.util.Map;

interface ProfileActivityView {
    void displayUserDefaults(Map<String, String> defaults);
    void showProgressBar();
    void hideProgressBar();
    void displayNoPreviousOrders();
    void displayPreviousOrders(List<PreviousOrder> orderList);
    void displayMessage(String message);
}

File: app/src/main/java/com/marknkamau/justjava/ui/profile/ProfileActivityView.kt
package com.marknkamau.justjava.ui.profile

import com.marknkamau.justjava.models.PreviousOrder
import com.marknkamau.justjava.models.UserDefaults

internal interface ProfileActivityView {
    fun displayUserDefaults(userDefaults: UserDefaults)
    fun showProgressBar()
    fun hideProgressBar()
    fun displayNoPreviousOrders()
    fun displayPreviousOrders(orderList: List<PreviousOrder>)
    fun displayMessage(message: String?)
}
